AI/Utilities/LifegainLogic.py

```
@@ -0,0 +1,181 @@
+
+_payoff_memo: dict[int, bool] = {}
+
+
+def _payoff_lookup(grp_id: int) -> bool:
+    if grp_id in PAYOFF_GRPIDS:
+        return True
+    if grp_id in _payoff_memo:
+        return _payoff_memo[grp_id]
+    try:
+        text = str(CardInfo.get_oracle_text(grp_id) or "")
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)